src/renderer/Theming.js

```diff
@@ -0,0 +1,77 @@
+const React = require('react')
+const StyledThemeProvider = require('styled-components').ThemeProvider
+const { EventEmitter } = require('events')
+
+const defaultTheme = Object.freeze({
+  deltaPrimaryBg: '#415e6b',
+  deltaPrimaryBgLight: '#4c6e7d',
+  deltaPrimaryFg: '#ffffff',
+  deltaPrimaryFgLight: '#d0d0d0',
+  deltaHover: '#ececec',
+  deltaChatPrimaryFg: '#070c14',
+  deltaChatPrimaryFgLight: '#62656a',
+  deltaChatMessageBubbleSelf: '#efffde',
+  deltaChatMessageBubbleOther: '#ffffff',
+  deltaInfoMessageBubbleBg: '#000000',
+  deltaInfoMessageBubbleColor: 'white',
+  deltaChatMessageBubbleSelfStatusColor: '#4caf50',
+  deltaFocusBlue: '#42A5F5',
+  deltaSelected: '#4c6e7d' // same as deltaPrimaryBgLight
+})
+
+class ThemeManager extends EventEmitter {
+  constructor () {
+    super()
+    const themeJSON = window.localStorage.getItem('theme')
+    this.currentTheme = themeJSON !== null ? JSON.parse(themeJSON) : {}
+
+    window.ThemeManager = this // only for using fron the dev console
+  }
+
+  setTheme (theme) {
+    window.localStorage.setItem('theme', JSON.stringify(theme))
+    this.currentTheme = theme
+    this.emit('update')
+  }
+
+  getCurrentTheme () {
+    return Object.assign({}, defaultTheme, this.currentTheme)
+  }
+}
+
+const manager = new ThemeManager()
+
+class ThemeProvider extends React.Component {
+  constructor () {
+    super()
+    this.state = {
+      theme: manager.getCurrentTheme()
+    }
+    this.update = this.update.bind(this)
+  }
+
+  componentDidMount () {
+    manager.addListener('update', this.update)
+  }
+
```

Comment:
Does it really need this much? Isn't something like (just an idea, no actual code) enough:
```
const lightTheme = {...}
const darkTheme = {...}

...
function ThemeProvider {
   const [selectedTheme, setSelectedTheme] = useState('light')
   dc.on('DC_SELECT_THEME', setSelectedTheme)
   return (<StyledThemeProvider theme={rc.theme === 'light' ? lightTheme : darkTheme}>
      {this.props.children}
    </StyledThemeProvider>)
}
```

Reply:
I'd like to have more than one theme / maybe custom themes. But doing it from the backend would be good for working with files.